PhysicsTools/NanoAOD/python/jetsAK4_Puppi_cff.py

```diff
@@ -111,6 +119,41 @@
 #jets are not as precise as muons
 jetPuppiTable.variables.pt.precision=10
 
+##############################################################
+## DeepInfoAK4:Start
+## - To be used in nanoAOD_customizeCommon() in nano_cff.py
+###############################################################
+from PhysicsTools.PatAlgos.tools.jetTools import updateJetCollection
+def nanoAOD_addDeepInfoAK4(process,addParticleNet):
+    _btagDiscriminators=[]
+    if addParticleNet:
+        print("Updating process to run ParticleNetAK4")
+        from RecoBTag.ONNXRuntime.pfParticleNetFromMiniAODAK4_cff import _pfParticleNetFromMiniAODAK4CHSCentralJetTagsAll as pfParticleNetFromMiniAODAK4CHSCentralJetTagsAll
+        from RecoBTag.ONNXRuntime.pfParticleNetFromMiniAODAK4_cff import _pfParticleNetFromMiniAODAK4CHSForwardJetTagsAll as pfParticleNetFromMiniAODAK4CHSForwardJetTagsAll
+        _btagDiscriminators += pfParticleNetFromMiniAODAK4CHSCentralJetTagsAll
+        _btagDiscriminators += pfParticleNetFromMiniAODAK4CHSForwardJetTagsAll
+    if len(_btagDiscriminators)==0: return process
+    print("Will recalculate the following discriminators: "+", ".join(_btagDiscriminators))
+    updateJetCollection(
+        process,
+        jetSource = cms.InputTag('slimmedJetsPuppi'),
+        jetCorrections = ('AK4PFPuppi', cms.vstring(['L2Relative', 'L3Absolute']), 'None'),
```

Comment:
Why not adding the L1 and L2L3 here? They are used above: https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jetsAK4_Puppi_cff.py#L12-L15

Reply:
True, L1 is not needed for Puppi and L2L3Res is only for data.
I'd also add a warning here. The current corrections won't be valid for the regressed jets, so why do we need them in the first place? I didn't check the code, but this should be valid for the mini step.

Reply:
Hi @anmalara, the idea was to make the ParticleNet regression correction relative to the raw jet pT available in the MINIAOD and NANOAOD even if you're right that the current JEC won't apply to the regressed jets. The idea was that we embed the ParticleNet outputs in the jet collections including the raw pT regression correction as something that can be studied to assess the performance gain from the resolution improvement even if one cannot use the regression as-is for an analysis for the time being. Then on the longer term we either integrate the regression as part of the JERC, as you know well, or we deploy an updated training with the regression relative to the JEC-corrected pT.

Reply:
This makes sense to me. I wanted to make clear (this information should be propagated carefully in twikis and it will) that the pt-regression and the pt-corrected cannot be mixed and if one wants to use the regression, one would need to reapply JEC on the fly.

Reply:
Thanks @anmalara , perhaps the L1FastJet could be removed here then? https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jetsAK4_Puppi_cff.py#L12

Reply:
do you have a specific reason to remove this correction? The JERC group would make sure that if L1 is not needed for Puppi, a dummy file is created. Should we decide that we need L1 for Puppi, we should revert these changes. Maybe, it's safer to include this correction even in this PR.
Thoughts?

Reply:
No that's fine then, if JERC would rather keep the dummy L1, we can leave it in, it's up to you.